src/Core/Configurations/RuntimeConfigProvider.cs

```diff
@@ -74,6 +76,34 @@ public RuntimeConfigProvider(RuntimeConfigLoader runtimeConfigLoader)
     /// <exception cref="DataApiBuilderException">Thrown when the loader is unable to load an instance of the config from its known location.</exception>
     public RuntimeConfig GetConfig()
     {
+        // Only used in hot reload to validate the configuration file
+        if (_configLoader.isNewConfigDetected && !_configLoader.isNewConfigValidated)
+        {
+            IFileSystem fileSystem = new FileSystem();
+            ILoggerFactory loggerFactory = new LoggerFactory();
+            ILogger<RuntimeConfigValidator> logger = loggerFactory.CreateLogger<RuntimeConfigValidator>();
+            RuntimeConfigValidator runtimeConfigValidator = new(this, fileSystem, logger, true);
+
+            _configLoader.isNewConfigDetected = false;
+            _configLoader.isNewConfigValidated = runtimeConfigValidator.TryValidateConfig(ConfigFilePath, loggerFactory).Result;
+
+            // Saves the lastValidRuntimeConfig as the new RuntimeConfig if it is validated for hot reload
+            if (_configLoader.isNewConfigValidated)
+            {
+                _configLoader.lastValidRuntimeConfig = _configLoader.RuntimeConfig;
+            }
+            else
+            {
+                _configLoader.isNewConfigValidated = true;
+                _configLoader.RuntimeConfig = _configLoader.lastValidRuntimeConfig;
+
+                throw new DataApiBuilderException(
+                    message: "Failed validation of configuration file.",
+                    statusCode: HttpStatusCode.ServiceUnavailable,
+                    subStatusCode: DataApiBuilderException.SubStatusCodes.ErrorInInitialization);
+            }
```

Comment:
how have we tested this? Is there a way to add an automated integration test in ConfigurationTests.cs?

Reply:
Tests were done manually, by using the placeholder schema url as an invalid configuration schema. I will see if there is a way to add automated integration tests.